### PR TITLE
Default empty target group NETCoreApp

### PR DESCRIPTION
**Pull request by ericstj**
This refactors all builds to make the netcoreapp-latest to be the
default build.

To do this I needed to be specific about the contract project in a few
cases so I made a fix to build tools to permit that and brought over
that fix as part of this commit.

In the future (once we stop building netstandard refs in corefx) we can make this directly enforced with the targets.

/cc @weshaggard @ellismg @chcosta

### CODE REVIEW

File: src/System.Collections/src/System.Collections.csproj [hunk 1]
```diff
@@ -20,6 +20,11 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netstandard1.7_Release|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcoreapp1.1_Debug|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcoreapp1.1_Release|AnyCPU'" />
+  <ItemGroup Condition="'$(TargetGroup)' == 'net463'">
+    <ContractProject Include="..\ref\System.Collections.csproj">
+      <TargetGroup>netstandard1.7</TargetGroup>
+    </ContractProject>
```
Comment (chcosta) [inline src/System.Collections/src/System.Collections.csproj]: I was confused by this usage, but I see this is new functionality for an older concept - https://github.com/dotnet/buildtools/commit/5da1a3e63c67e12ae5ab4073c9ac289e1713e86e

What does this do exactly?  Does it ensure that there is a project reference to the reference assembly project?  Are there scenarios where we actually have to define the OSGroup for these?

File: src/System.Buffers/src/System.Buffers.csproj [hunk 2]
```diff
@@ -6,20 +6,20 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <DocumentationFile>$(OutputPath)$(AssemblyName).xml</DocumentationFile>
     <UseOpenKey Condition="'$(UseOpenKey)'==''">true</UseOpenKey>
-    <IsPartialFacadeAssembly Condition="'$(TargetGroup)' == 'netcoreapp1.1'">true</IsPartialFacadeAssembly>
-    <NuGetTargetMoniker Condition="'$(TargetGroup)' == ''">.NETStandard,Version=v1.1</NuGetTargetMoniker>
+    <IsPartialFacadeAssembly Condition="'$(TargetGroup)' == ''">true</IsPartialFacadeAssembly>
+    <NuGetTargetMoniker Condition="'$(TargetGroup)' == ''">.NETCoreApp,Version=v1.1</NuGetTargetMoniker>
```
Comment (chcosta) [inline src/System.Buffers/src/System.Buffers.csproj]: Is there a reason not to centrally define the latest values (netcoreapp1.1 and netstandard1.7 plus monikers) and then override them if necessary in the build projects?  Might make rolling forward in the future easier.

**Comment by ericstj** [comment]
This is breaking because of another breaking change in buildtools: https://github.com/dotnet/buildtools/commit/2ba3679dbccd7ee0a79ae8d18fede36232417aff#diff-c368ca147f7a6d95f33280f334ff9e84R109

/cc @karajas

**Comment by weshaggard** [comment]
Is this complete because it seems small? I suppose you have only flipped the default for any projects that already have a specific netcoreapp1.1 build. Will the next step be to add that configuration for all the projects?

**Comment by ericstj** [comment]
Right now this is just making it true that a filter for TargetGroup='' and whatever OSGroup is appropriate will give you all of the NETCoreApp assets needed for all packages.

**Comment by ericstj** [comment]
This is going to fail due to https://github.com/dotnet/buildtools/commit/2ba3679dbccd7ee0a79ae8d18fede36232417aff#commitcomment-19651092, putting a fix in buildtools.

**Comment by ericstj** [comment]
@alexperovich the latest failure looks like it is code-analysis related.
```
CSC : error CS8032: An instance of analyzer Desktop.Analyzers.CSharpDoNotUseInsecureCryptographicAlgorithmsAnalyzer cannot be created from /mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/Tools/analyzers/Desktop.CSharp.Analyzers.dll : Could not load file or assembly 'Desktop.Analyzers, Version=1.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified. [/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/src/Microsoft.CSharp/src/Microsoft.CSharp.csproj]
```
Looks like its happening on all unix builds.

**Comment by ericstj** [comment]
Seems to be a difference between desktop and core.  I ran core CSC on a windows machine and repro'ed the same issue.  It looks like Roslyn's AssemblyLoadContext doesn't look next to the analyzer, nor in the list of analyzers when trying to find dependencies.  /cc @tmat

**Comment by tmat** [comment]
@tmeschter @mavasany to comment on the analyzer issue

**Comment by ericstj** [comment]
Worked around that issue: https://github.com/dotnet/buildtools/pull/1185.  Now hopefully will have a clean build 🙏

**Comment by tmeschter** [comment]
The loader used by core intentionally doesn't look next to an analyzer to find dependencies; it only checks the files in the list of analyzers. It looks like we have a bug in that logic.

On desktop the compiler is a little more forgiving since we just use `LoadFrom`, and it will look next to loaded assemblies for dependencies. This isn't the desired behavior so much as it is an unavoidable one on desktop.

**Comment by tmeschter** [comment]
@ericstj Could you detail how you repro'd this on Windows?

**Comment by ericstj** [comment]
@tmeschter: see https://github.com/dotnet/roslyn/issues/14866.  Thanks for opening @alexperovich

**Pull request merged**